platforms/documentation/docs/src/snippets/bestPractices/avoidDuplicateDependencies/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,22 @@
+plugins {
+    `java-library`
+}
+
+// tag::avoid-this[]
+dependencies {
+    implementation("org.jetbrains:kotlinx-coroutines-core:1.10.0") // <1>
+    // ...
+    // long dependencies declaration list continues
+    // ...
+    implementation("org.jetbrains:kotlinx-coroutines-core:1.6.0") // <2>
+}
+// end::avoid-this[]
+
+// tag::do-this[]
+dependencies {
+    implementation("org.jetbrains:kotlinx-coroutines-core:1.10.0") // <1>
+}
+// end::do-this[]
+
+// dummy task to be used in tests
+tasks.register("dummyTask")
```

Comment:
I'd organize this into 2 builds and run `dependencies` on each of them, showing that the resolved deps are equivalent.

Reply:
We should have a `do-this` and a `don't-do-this` build. A simple example would be:

Don't do this:
```
dependencies {
    implementation("org.jetbrains:kotlinx-coroutines-core:1.10.0")
    api("org.jetbrains:kotlinx-coroutines-core:1.10.0")
}
```

Do this:
```
dependencies {
    implementation("org.jetbrains:kotlinx-coroutines-core:1.10.0")
}
```

Reply:
Ok! I'll re-write my example to "implementation/api" redundancy case.

Still, I'm interested in showing the problem with version conflicts as it is more painful and I've seen it irl.

Reply:
> Still, I'm interested in showing the problem with version conflicts as it is more painful and I've seen it irl.

It would be ideal to showcase a more "interesting" example. 

Also can you update the examples to have two subprojects (`do-this` and `dont-do-this`) and make sure they are tested. For example: https://github.com/gradle/gradle/pull/32825/files

Reply:
> Also can you update the examples to have two subprojects (do-this and dont-do-this) 

This is agreed convention? I like it!
I'll update samples to this layout.

Reply:
Updated, please re-check.